pennylane/_qubit_device.py

```diff
@@ -629,14 +629,17 @@ def statistics(
                 result = self.var(obs, shot_range=shot_range, bin_size=bin_size)
 
             elif isinstance(m, SampleMP):
-                samples = self.sample(obs, shot_range=shot_range, bin_size=bin_size, counts=False)
+                sample_input = m if m.mv is not None else obs
```

Comment:
oh I see, you're setting it to `m`, not `m.mv`, so this is actually a change. hmm yeah, if you can make it more evident what's happening, it would be appreciated. but given that it's actually a necessity, I'm ok with it however you get it done.

Reply:
Updated :)